pkg/conversion/mapi2capi/powervs.go

```diff
@@ -0,0 +1,371 @@
+	}
+
+	return &capibmv1.IBMPowerVSMachine{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: capibmv1.GroupVersion.String(),
+			Kind:       ibmPowerVSMachineKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      m.machine.Name,
+			Namespace: capiNamespace,
+		},
+		Spec: spec,
+	}, errs
+}
+
+// fromPowerVSMAPIMachineToCAPIMachine translates a PowerVS MAPI Machine to its Core CAPI Machine correspondent.
+// TODO: May be generalise to support different infrastructure machines.
+func fromPowerVSMAPIMachineToCAPIMachine(mapiMachine *mapiv1beta1.Machine) (*capiv1.Machine, field.ErrorList) {
+	var errs field.ErrorList
+
+	capiMachine := &capiv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        mapiMachine.Name,
+			Namespace:   capiNamespace,
+			Labels:      mapiMachine.Labels,
+			Annotations: mapiMachine.Annotations,
+			// OwnerReferences: TODO(OCPCLOUD-2716): These need to be converted so that any MachineSet owning a Machine is represented with the correct owner reference between the two APIs.
+		},
+		Spec: capiv1.MachineSpec{
+			InfrastructureRef: corev1.ObjectReference{
+				APIVersion: ibmPowerVSMachineAPIVersion,
+				Kind:       ibmPowerVSMachineKind,
+				Name:       mapiMachine.Name,
+				Namespace:  capiNamespace,
+			},
+			ProviderID: mapiMachine.Spec.ProviderID,
+
+			// Version: TODO(OCPCLOUD-2714): To be prevented by VAP.
+			// ClusterName: populated by higher level functions.
+
+			// TODO(OCPCLOUD-2715): These are not present on the MAPI API, we should implement them for feature parity.
+			// NodeDrainTimeout: ,
+			// NodeVolumeDetachTimeout: ,
+			// NodeDeletionTimeout: ,
+		},
+	}
+
+	// lifecycleHooks are handled via an annotation in Cluster API.
+	lifecycleAnnotations := getCAPILifecycleHookAnnotations(mapiMachine.Spec.LifecycleHooks)
+	if capiMachine.Annotations == nil {
+		capiMachine.Annotations = lifecycleAnnotations
+	} else {
+		for key, value := range lifecycleAnnotations {
+			capiMachine.Annotations[key] = value
+		}
+	}
+
+	if capiMachine.Labels == nil {
+		capiMachine.Labels = map[string]string{}
+	}
+
+	errs = append(errs, setMAPINodeLabelsToCAPIManagedNodeLabels(field.NewPath("spec", "metadata", "labels"), mapiMachine.Spec.ObjectMeta.Labels, capiMachine.Labels)...)
+
+	// Unused fields - Below this line are fields not used from the MAPI Machine.
+
+	if len(mapiMachine.OwnerReferences) > 0 {
+		// TODO(OCPCLOUD-2716): We should support converting CAPI MachineSet ORs to MAPI MachineSet ORs. NB working out the UID will be hard.
+		errs = append(errs, field.Invalid(field.NewPath("metadata", "ownerReferences"), mapiMachine.OwnerReferences, "ownerReferences are not supported"))
+	}
+
+	// mapiMachine.Spec.AuthoritativeAPI - Ignore as this is part of the conversion mechanism.
+
+	// metadata.labels - needs special handling
+	// metadata.annotations - needs special handling
+
+	errs = append(errs, handleUnsupportedMachineFields(mapiMachine.Spec)...)
+
+	return capiMachine, errs
+}
```

Comment:
Apart from the APIVersion and Kind, everything here looks identical to the AWS version, can we deduplicate this easily in a new file?

Reply:
Created util file and moved it there with kind as a parameter to identify the different types of machines.